packages/tupaia-web/src/layout/MainLayout.tsx

```diff
@@ -31,7 +31,24 @@ const Container = styled.div`
   }
 `;
 
+// on mount, if the url is just the explore page, redirect to the projects tab. This only happens on first mount of this view. This needs to go here, not in ProjectPage on the off chance a custom landing page directs to the explore project
+const useShowProjectsModal = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  useEffect(() => {
+    if (
+      (location.pathname === DEFAULT_URL || location.pathname === DEFAULT_PROJECT_ENTITY) &&
+      !location.hash
+    ) {
+      navigate({
+        ...location,
+        hash: MODAL_ROUTES.PROJECTS,
+      });
+    }
+  }, []);
+};
```

Comment:
I was on the fence about whether this should be inside the component or outside, but went with outside because it makes the component less busy to read

Reply:
I think this is not quite right because it will only work if the user refreshed on to the home page and in the case of logging in they have moved there so I think this should be moved to the login success handler instead. eg.

```
if (location.state?.referrer) {
    navigate(location.state.referrer, {
      state: null,
    });
  } else if (
    location.pathname === DEFAULT_URL ||
    location.pathname === DEFAULT_PROJECT_ENTITY
  ) {
    navigateToModal(MODAL_ROUTES.PROJECTS);
  } else {
    closeModal();
 }
}

Reply:
How about when the user is already logged in? Then it won't happen?